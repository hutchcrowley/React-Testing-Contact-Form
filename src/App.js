import React from 'react'

import ContactForm from './components/ContactForm'

export default function App() {
	return (
		<div className='App' data-testid='app'>
			<h1 className='app-h1' data-testid='app-h1'>
				Unit-Testing React Apps
			</h1>
			<ContactForm />
		</div>
	)
}
