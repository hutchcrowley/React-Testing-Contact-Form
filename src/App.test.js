import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import App from './App'

test('app renders successfully', async () => {
	const { getByTestId } = render(<App />)
	expect(getByTestId('app')).toBeInTheDocument()
})

test('app renders successfully', async () => {
	const { getByText } = render(<App />)
	expect(getByText('Unit-Testing React Apps')).toBeVisible()
})

test('this test should fail', () => {
	const { getByText } = render(<App />)
	expect(getByText('Unit-Testing React Apps')).not.toBeVisible()
})

test('snapshot is unchanged', async () => {
	const { getByTestId } = render(<App />)

	const tree = getByTestId('app')
	expect(tree).toMatchSnapshot()
})
