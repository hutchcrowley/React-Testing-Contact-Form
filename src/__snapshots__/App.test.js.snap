// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot is unchanged 1`] = `
<div
  class="App"
  data-testid="app"
>
  <h1
    class="app-h1"
    data-testid="app-h1"
  >
    Unit-Testing React Apps
  </h1>
  <div
    class="App"
  >
    <form
      class="contact-form"
      data-testid="contactForm"
    >
      <div
        data-testid="FNInput"
      >
        <label
          for="firstName"
        >
          First Name*
        </label>
        <input
          name="firstName"
          placeholder="First Name"
        />
      </div>
      <div
        data-testid="lnInput"
      >
        <label
          for="lastName"
        >
          Last Name*
        </label>
        <input
          name="lastName"
          placeholder="Last Name"
        />
      </div>
      <div
        data-testid="eInput"
      >
        <label
          for="email"
        >
          Email*
        </label>
        <input
          name="email"
          placeholder="email@somesite.com"
        />
      </div>
      <div
        data-testid="mInput"
      >
        <label
          for="message"
        >
          Message
        </label>
        <textarea
          name="message"
        />
      </div>
      <button
        class="btn"
        data-testid="sBtn"
        type="submit"
      >
        SUBMIT FORM
      </button>
      <button
        class="btn"
        data-testid="rBtn"
      >
        RESET FORM
      </button>
    </form>
  </div>
</div>
`;
